helm/myapp-chart/templates/istio/destinationrules.yml

```diff
@@ -0,0 +1,53 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: {{ .Release.Name }}-frontend-dr
+spec:
+  host: {{ .Release.Name }}-app-frontend-service
+  trafficPolicy:
+    loadBalancer:
+      consistentHash:
+        useSourceIp: true
+  subsets:
+  - name: v1
+    labels:
+      version: v1
+  - name: v2
+    labels:
+      version: v2
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: {{ .Release.Name }}-app-service-dr
+spec:
+  host: {{ .Release.Name }}-app-service
+  trafficPolicy:
+    loadBalancer:
+      consistentHash:
+        useSourceIp: true
+  subsets:
+  - name: v1
+    labels:
+      version: v1
+  - name: v2
+    labels:
+      version: v2
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: {{ .Release.Name }}-model-service-dr
+spec:
+  host: {{ .Release.Name }}-model-service
+  trafficPolicy:
+    loadBalancer:
+      consistentHash:
+        useSourceIp: true
+  subsets:
+  - name: v1
+    labels:
+      version: v1
+  - name: v2
+    labels:
+      version: v2 
```

Comment:
We don't have versioned deployments for app-service and model-service, do we?
I'm a bit confused why we are passing destination rules for their subsets... do we need to make them versioned?

Reply:
Update: This is wrong because the deployments don't contain any label called "version: v1" or "version: v2" so Istio gateway cannot route requests correctly.